src/ctelnet.cpp

```diff
@@ -2175,9 +2173,16 @@ void cTelnet::handle_socket_signal_readyRead()
 
     char in_bufferx[100010];
     char* in_buffer = in_bufferx;
-    char out_buffer[100010];
 
     int amount = socket.read(in_buffer, 100000);
+    readSocket(in_buffer, amount);
+}
+
+void cTelnet::readSocket(char*& in_buffer, int amount) {
+    mpHost->mInsertedMissingLF = false;
```

Comment:
If it is revised to bail out on error then this will need to go back in before that point.

Reply:
Does it? The old code did not.

Reply:
I meant, if you went with - what *I* think is a better arrangement - to **not** call `readSocket(...)` if `amount` is negative (i.e. in an error condition) then this line would need to be executed back in `handle_socket_signal_readyRead(...)` first.

Reply:
Why is it a better arrangement? The function already checks if the `amount` is negative. Why check twice?

Reply:
Well if we change the confusing name (`readSocket` is not reading the socket - that has already been done before it is called - perhaps it should be `processSocketData` :wink: ) then does **not** executing `processSocketData` should `amount` have the sentinel `-1` (or indeed *any* negative value) seem like a more sane approach.

Given that `amount` refers to the size of a `char` array then using the `size_t` (which, being intended for indexing into arrays is the correct) type to quantify it in the call to what I am calling `processSocketData`. By checking `amount` to be a non-negative you have bounds checked that it can be `static_cast<size_t>(...)` from the `qint64` type that `socket.read(...)` returns into the argument that `processSocketData` wants; there is *no* need to then check it and `return` prematurely from *that* method...

Reply:
Fuzzing works by generating random inputs - so we want to check the size of our inputs within `processSocketData`, and and not another funciton, because fuzzing will be calling `processSocketData` and not another function.